### PR TITLE
Fix ddp pruner problem

### PR DESCRIPTION
**Pull request by Bonytu**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### How to test ###
use taylor and activationbased pruners to compress ddp model, then check that masks are the same on different gpus.

### CODE REVIEW

File: nni/algorithms/compression/v2/pytorch/base/compressor.py [hunk 1]
```diff
@@ -103,6 +105,38 @@ def reset(self, model: Module, config_list: List[Dict]):
...
         self._wrap_model()
 
+    def check_ddp_model(self):
+        if isinstance(self.bound_model, torch.nn.parallel.DistributedDataParallel):
+            attr_dicts = self.bound_model.__dict__
+
+            if "device_ids" in attr_dicts:
+                self.ddp_params = {"device_ids": attr_dicts.get("device_ids", None)}
+            if "output_device" in attr_dicts:
+                self.ddp_params["output_device"] = attr_dicts.get("output_device", None)
```
[the remark this answers is not in the document]
Comment (Bonytu) [reply nni/algorithms/compression/v2/pytorch/base/compressor.py:115]: because different pytorch version may have different params
